### ISSUE TITLE
Internationalization

### ISSUE THREAD

**Issue by eteran** [eteran/nedit-ng#4]
Using Qt makes translation possible. If there is anyone who is able to provide translations, they will be welcome!

**Comment by ericxuo** [comment]
Hi,

I can help for french translation. But I need you explain me how to proceed.

Regards.

Xuo.

**Comment by eteran** [comment]
Awesome, that is great news. I just need to figure out how to add the necessary parts to the cmake build system.

**Comment by eteran** [comment]
OK, so here's the first steps (assuming "fr" for french)

1. install Qt linguist
2. cd into the "source" directory of nedit-ng
3. run `lupdate *.cpp *.h *.ui -ts res/nedit-ng_fr.ts`. This will produce the XML file for translation
4. run `linguist res/nedit-ng_fr.ts`
5. Once this is complete, submit a PR to add this file to the repo, at which point I will add "compiling" it to the build process and will be able to test loading the translation file at runtime.

There's a lot of strings, so don't feel like you need to get them all done in one shot! Thanks!

**Comment by ericxuo** [comment]
Hi,

I prefer to ask as I'm not 100% sure of what should be done.

![calltipwidget](https://user-images.githubusercontent.com/9089499/36938609-5f285596-1f24-11e8-9b50-e9e127061c3f.png)

Here, I think there is nothing to translate.

![commandrecorder](https://user-images.githubusercontent.com/9089499/36938610-6d4b5696-1f24-11e8-9df4-14a149d2a249.png)
Here neither ...

![dialogabout](https://user-images.githubusercontent.com/9089499/36938646-f046e1dc-1f24-11e8-9ea5-8129cb01f0f5.png)

Here, I should copy/paste all the original html text in the "Français traduction" field and translate the useful parts.

Do you agree ?

Regards.

Xuo.

**Comment by eteran** [comment]
Yes, for fields which have nothing to translate, it should simply have the original text. I **think** that if you mark an item as complete (the toolbar button with the green check) without putting anything in the "Français traduction" field, then it'll default to the original text... We should check to be sure though.

**Comment by eteran** [comment]
Yup just checked the manual: "Missing translations will be replaced with the native language phrases at runtime". So if there is literally nothing to translate, just mark it as complete :-)

**Comment by 1div0** [comment]
lupdate-qt5 on F27

**Comment by ericxuo** [comment]
Hi,

I don't understand how to open a PR. I can add the .ts file here as an attachment and you can include it where you want.
You agree ?

Regards.

Xuo.

**Comment by eteran** [comment]
Sure, that'll be fine 🙂

**Comment by ericxuo** [comment]
Hi,

I don't understand how to open a PR. I can add the .ts file here as an attachment and you can include it where you want.
You agree ?

Regards.

Xuo.

**Comment by ericxuo** [comment]
Hi,

Here it is (I renamed it from nedit-ng_fr.ts to nedit-ng_fr.txt as the .ts extension is not allowed here).

[nedit-ng_fr.txt](https://github.com/eteran/nedit-ng/files/1778962/nedit-ng_fr.txt)

Regards.

Xuo.

**Comment by ericxuo** [comment]
Hi,

Another question : 
Sometime, there is something like that to translate : 
&Regular expression
which make the R to be underlined which maybe means that 'R' could be an accelerator to tick the dialog box corresponding to the field 'Regular Expression'.

In french, it will be translated to : 
Expression régulière
Should I write : 
Expression régulière
or 
&Expression régulière
or 
Exp&Ression régulière
or
Exp&ression régulière

Regards.

Xuo.

**Comment by ericxuo** [comment]
Hi,

Here is the next version.
[nedit-ng_fr_v2.txt](https://github.com/eteran/nedit-ng/files/1782402/nedit-ng_fr_v2.txt)

Regards.

Xuo.

**Comment by eteran** [comment]
Wonderful! I'll start working on the code to compile and include translations 👍

**Comment by eteran** [comment]
OK, I'm about to push a version that should use your translations (found in `source/res/translations/nedit-ng_fr.ts`). Easily testable with:

    LANGUAGE=fr ./nedit-ng

A few notes:

1. Thank you for the effort!
2. I don't think we should translate the GPL though, let's leave that as is.
3. Try to pay attention to things like whitespace. In the about dialog, it uses a fixed width font and the English is lined up to look nice; please attempt to maintain that in the translations.
4. Likewise, be careful about introducing new leading or trailing whitespace. The UI actually warns about this (yellow checkmark and message in the warnings pane)
5. What name would you like me to add to the translator's list in the about dialog?
6. Thank you again!

**Comment by ericxuo** [comment]
Hi,

Yes, I know I didn't respect whitespace, semicolons, ... This is because I prefer, for ex, to add a space before a colon. I've noticed the yellow stuff but I though it was not important.
I'll change this to get everything green (and even dark green).
About the name, I just don't care. Xuo will be fine.

Regards.

Xuo.

**Comment by eteran** [comment]
BTW, now that the file is in the repo, please look into how to do a PR.

Basically, the process is:

1. fork the repo
2. create a branch in your version of the repo
3. make changes as you see fit to your version
4. in github, you can ask it to create a PR to merge your changes with my repo
5. make sure your repo remains in sync with the main one (only complicated part I think)

See this page for more in depth details: https://help.github.com/articles/creating-a-pull-request/

**Comment by ericxuo** [comment]
Hi,

I had translated everything. This is not what you see ?
Except maybe if you have added some new stuff since my last PR.

Regards.

Xuo.

**Comment by eteran** [comment]
Oh your translation seemed great. but some strings have changed slightly, some were removed, and a few added, so it could use a refresh :-)

**Comment by ericxuo** [comment]
Hi,

OK. I'll have a look. I just need to remember how I did. I think I've lost all my previous nedit-ng translation configuration (change of PC). I think I've kept somewhere a "HowTo".

Regards.

Xuo.

**Comment by Barracuda1984** [comment]
Hi 
I can make a Czech translation.
Regards 

Zdenek (Barracuda)

**Comment by eteran** [comment]
That would be fantastic!
